DigitalLearningSolutions.Data/Services/SupervisorService.cs

```diff
@@ -567,25 +567,46 @@ public bool RemoveCandidateAssessment(int candidateAssessmentId)
             return true;
         }
 
-        public bool RemoveCandidateAssessmentSupervisor(int selfAssessmentId, int supervisorDelegateId)
-        {
-            var numberOfAffectedRows = connection.Execute(
-         @"DELETE FROM cas
-FROM  CandidateAssessmentSupervisors AS cas INNER JOIN
-         CandidateAssessments AS ca ON cas.CandidateAssessmentID = ca.ID
-WHERE (ca.SelfAssessmentID = @selfAssessmentId) AND (cas.SupervisorDelegateId = @supervisorDelegateId)",
-        new { selfAssessmentId, supervisorDelegateId });
-            if (numberOfAffectedRows < 1)
+        public bool RemoveCandidateAssessmentSupervisor(int supervisorDelegateId)
+        {
+            var numberOfRemovedVerifications = connection.Execute(
+                @"DELETE FROM sarsv

```

Comment:
This is hard deleting SelfAssessmentResultSupervisorVerifications - we should never do that. The implementation should be checking if they exist (or if any related CandidateAssessmentSupervisorVerifications exist) and soft deleting the CandidateAssessmentSupervisors record if they do.

We then need to check queries called from the LearningPortal/SelfAssessments controllers and the Supervisor controllers to make sure they exclude soft deleted CandidateAssessmentSupervisors records.